spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java

```diff
@@ -376,6 +388,19 @@ private void doIfTransaction(Context ctx, Consumer<ReactiveFirestoreResourceHold
 		}
 		String resourceName = buildResourceName(entity);
 		Document document = getClassMapper().entityToDocument(entity, resourceName);
+		FirestorePersistentEntity<?> persistentEntity =
+				this.mappingContext.getPersistentEntity(entity.getClass());
+		FirestorePersistentProperty updateTimeProperty = persistentEntity.getUpdateTimeProperty();
+		if (updateTimeProperty != null && updateTimeProperty.findAnnotation(UpdateTime.class).version()) {
+			Object version = persistentEntity.getPropertyAccessor(entity).getProperty(updateTimeProperty);
+			if (version != null) {
+				builder.setCurrentDocument(
+						Precondition.newBuilder().setUpdateTime(((Timestamp) version).toProto()).build());
+			}
+			else {
+				builder.setCurrentDocument(Precondition.newBuilder().setExists(false).build());
```

Comment:
Why should we prevent the save if the item exists? How is it related to optimistic locking?

Reply:
If you try to save an entity with an empty update time field, it must be new. Otherwise you will overwrite an existing document.

Reply:
Got it. Maybe some in-line comments would help here.